src/ConfigParser.cpp

```diff
@@ -1,14 +1,103 @@
 #include "../include/ConfigParser.hpp"
+#include "../include/Server.hpp"
+#include "ConfigParser.hpp"
 
-//ConfigParser::ConfigParser() : lineCount(0), start(0)
-//{
-//}
+void ConfigParser::parseConfig(const std::string& filename) 
+{
+	std::ifstream file(filename);
+	std::string line;
 
-//ConfigParser::~ConfigParser()
-//{
-//}
+	if (!file.is_open()) 
+	{
+		std::cerr << "Failed to open file: " << filename << std::endl;
```

Comment:
throw a ConfigurationException here instead of writing to cerr and exiting, we already handle that in the main function!